design/slot-reservations.md

```diff
@@ -0,0 +1,225 @@
+becomes available to be filled (or repaired) by allowing only a very small
+number of SP addresses to fill/repair the slot at the start. Over time, the
+number of eligible SP addresses increases, until eventually all SP addresses in
+the network are eligible.
+
+The expanding window mechanismstarts off with a random source address, defined
+as $hash(block hash, request id, slot index, reservationindex)$. The distance
+between each SP address and the source address can be defined as $XOR(A, A_0)$
+(kademlia distance). Once the allowed distance is greater than SP's distance,
+the SP is considered eligible to reserve a slot. The allowed distance for
+eligible addresses over time $t_i$ can be [defined as $2^{256} *
+F(t_i)$](https://hackmd.io/@bkomuves/BkDXRJ-fC). As this allowed distance value
+increases along a curve, more and more addresses will be eligible to participate
+in reserving that slot. In total, eligible addresses are those that satisfy:
+
+$XOR(A, A_0) < 2^{256} * F(t_i)$
+
+Because the source address for the expanding window is generated using the slot
+index and the reservation index, that means the source address for each
+reservation in each slot will be different.
+
+Furthermore, the client can set the rate of expansion by defining the [parameter
+$h$](https://hackmd.io/@bkomuves/BkDXRJ-fC#Parametrizing-the-speed-of-expansion)
+as the [dispersal parameter](https://github.com/codex-storage/codex-research/blob/ad41558900ff8be91811aa5de355148d8d78404f/design/marketplace.md#dispersal) of the request.
+Changing the value of $h$ will [affect the curve of the rate of
+expansion](https://www.desmos.com/calculator/pjas1m1472) (interactive graph).
+
+### Abandoned ideas
+
+#### No reservation collateral
+
+Reservation collateral was thought to be able to prevent a situation where an SP
+would reserve a slot then fail to fill it. However, collateral could not be
+burned as it created an attack vector for clients: clients could withhold the
+data and cause SPs to lose their reservation collateral. The reservation
+transaction itself creates a signal of intent from an SP to fill the slot. If
+the SP were to not fill the slot, then other SPs that have reserved the slot
+will fill it.
+
+#### No reservation reward
+
+Adding a fill reward for SPs who ultimately fill the slot is not necessary
+because, like the SP rewards for providing proofs, fill rewards would be paid
+once the storage request successfully completes. This would mean that the fill
+reward is effectively the same as an increase in value of the storage request
+payout. Therefore, if a client is so inclined to provide a fill reward, they
+could instead increase the total reward of the storage request.
+
+In this simplified slot reservations proposal, there will not be reservation
+collateral and reward requirements until the behavior in a live environment can
+be observed and it is determined this are necessary mechanisms.
```

Comment:
Also we said that the "get payed for actual time hosting the slot" would also align with the "fill reward" even though most probably not in its full "force".

See: https://github.com/codex-storage/codex-contracts-eth/issues/102

Reply:
Added additional comments to that effect.